tests/incomplete_parse_file.c

```diff
@@ -0,0 +1,77 @@
+#include "test.h"
+
+// Note this test may only return the correct result
```

Comment:
@KaruroChori Do you know of any ways to make sure there aren't leaks or do you think this should be sufficient? I have '-Db_sanitize=address,undefined -Db_lundef=false' appended to the meson setup command in the CI and locally.

Reply:
I am not really knowledgeable on how meson achieves that, but I think it just uses compiler flags.  
I always test with valgrind on the final binaries, but I never used those features directly offered by the compiler, as they are not universally present come with some build-time headaches.
So I really don't know :D.

Reply:
No problem, I've found using Db_sanitize=address,undefined is pretty reliable so far. I've seen it throw errors pretty frequently when a program otherwise wouldn't (when compiled without sanitize).

I just gave it a little manual test. I commented out L138

```
      // free((*node)->attributes);
      (*node)->attributes = NULL;
```

and most of the tests fail when sanitize is enabled, but they all pass otherwise.

I guess I'm comfortable it's good enough for now and can revisit later if anyone reports an actual problem.